### DISCUSSION TITLE
[Deployment] Need help with error while deploying contract

### DISCUSSION THREAD

**Discussion by pspiro** [zkSync-Community-Hub/zksync-developers#671] [Deployments] [answered]
### Environment

Testnet

### L2 block number

n/a

### zkSolc Version

Solidity compiler for ZKsync v1.5.2 (LLVM build 280f38e3b73cbed0b037aa7e6c073d8e25dbdaa1)

### zksync-ethers Version

using Zksync Web3j v0.2.0

### Minimal Reproducible Code or Repo Link

pragma solidity ^0.8.4;

import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
import "@openzeppelin/contracts/access/Ownable.sol";
import "@openzeppelin/contracts/security/ReentrancyGuard.sol";

contract BUSD is ERC20, Ownable, ReentrancyGuard {

    constructor() ERC20("Ref BUSD", "BUSD") {
    }

    function mint(address to, uint256 _amount) public {
        _mint(to, _amount);
    }

    function burn(address from, uint256 _amount) public {
        _burn(from, _amount);
    }

    function decimals() public pure override returns (uint8) {
        return 6;
    }
}

### Does this work on other EVMs? (If yes, please list at least 1 of them)

no

### Description of What Your Contract Does

Nothing, I am test-deploying a simple ERC-20 token with no added functionality

### Additional Details

I have the solc compiler installed:
`Version: 0.8.17+commit.8df45f5f.Windows.msvc`

I have the zksolc copiler installed:
`Solidity compiler for ZKsync v1.5.2 (LLVM build 280f38e3b73cbed0b037aa7e6c073d8e25dbdaa1)`

I compile my test contract with this command:
`zksolc --bin contracts/Busd.sol --base-path . --include-path node_modules/ -o build/`

I get the following output
`Compiler run successful. Artifact(s) can be found in directory "build/".`

I get the attached output for busd.zbin (compressed to busd.zip)
[BUSD.zip](https://github.com/user-attachments/files/16754366/BUSD.zip)

PROBLEM: when I deploy the bytecode, I get an error saying the bytecode length in bytes must be divisible by 32. (Shaving off the "0x" at the beginning does not help.)

Can anyone shed light on this for me?

**Comment by alhenaai** [comment]
It appears that the error you're encountering is related to the specific requirements for deploying contracts on ZKsync. According to the ZKsync documentation, the bytecode for a contract must meet certain criteria:

1. **Bytecode Length**: The length of the bytecode in bytes must be divisible by 32. This is because the bytecode is interpreted in 32-byte words.
2. **Odd Length in Words**: The length of the bytecode in 32-byte words must be odd.

Here are a few steps you can take to resolve this issue:

- **Check Bytecode Length**: Ensure that the length of your bytecode, excluding the "0x" prefix, is divisible by 32. If it's not, you may need to adjust your contract or its compilation settings.
- **Ensure Odd Length in Words**: After confirming the divisibility by 32, check that the number of 32-byte words (i.e., the total length divided by 32) is an odd number. If it's even, you might need to add or remove some operations or data in your contract to change the length.

For more detailed information on these requirements, you can refer to the ZKsync documentation on [Contract Deployment](https://docs.zksync.io/build/developer-reference/ethereum-differences/contract-deployment).

If adjustments to the contract do not resolve the issue, consider reviewing or seeking further assistance with the specifics of your contract's compilation and deployment process to ensure compatibility with ZKsync's requirements.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by bxpana** [comment]
Are you using the zksync-cli or just doing `forge init` to create a project and then trying to deploy the BUSD contract?

**Comment by uF4No** [comment, marked as answer]
Smart contract deployment is different on ZKsync vs EVM networks. you can [learn more in this section of the docs](https://docs.zksync.io/build/developer-reference/ethereum-differences/contract-deployment), but in short, in ZKsync you interact with a system contract called ContractDeployer.

As you're using Java, make sure you're using the the zksync java SDK: https://sdk.zksync.io/java/introduction/overview

**Comment by petarTxFusion** [comment]
Her you can find examples on how to [deploy contracts](https://github.com/zksync-sdk/zksync2-examples/tree/main/java/src/main/java/org/example/deploy). If you still experience issue could you please share your code
